docker/docker_image_dest.go

```diff
@@ -154,7 +154,11 @@ func (d *dockerImageDestination) PutBlob(ctx context.Context, stream io.Reader,
 	defer res.Body.Close()
 	if res.StatusCode != http.StatusAccepted {
 		logrus.Debugf("Error initiating layer upload, response %#v", *res)
-		return types.BlobInfo{}, errors.Wrapf(client.HandleErrorResponse(res), "Error initiating layer upload to %s in %s", uploadPath, d.c.registry)
+		err = client.HandleErrorResponse(res)
+		if e, ok := errors.Cause(err).(*client.UnexpectedHTTPResponseError); ok {
+			return types.BlobInfo{}, errors.Wrapf(fmt.Errorf("%s, StatusCode: %d", string(e.Response), e.StatusCode), "Error initiating layer upload to %s in %s", uploadPath, d.c.registry)
```

Comment:
Perhaps start with the status code, when `e.Response` can be arbitrarily large. “HTTP status %d: %s” or something like that.

This can be further simplified to avoid duplicating the “Error initiating layer upload” text with `err = fmt.Errorf(…)` and letting the code continue.